### DISCUSSION TITLE
Captive Dependency -  ValidateOnBuild not working

### DISCUSSION THREAD

**Discussion by rahman3593** [dotnet/aspnetcore#53109] [General]
<img width="1716" alt="Screenshot 2024-01-03 at 3 53 55 PM" src="https://github.com/dotnet/aspnetcore/assets/72451246/b14185bc-25cf-4e12-9a5a-63a17359cb20">

I was injecting scoped service in singleton service as shown in the above code in image. and I want the solution build to be failed because I have set the ValidateOnBuild property to true. But the build happens successfully.

Note: But when I try to run the application after build it is throwing error as below.
<img width="1698" alt="Screenshot 2024-01-03 at 4 03 43 PM" src="https://github.com/dotnet/aspnetcore/assets/72451246/4c6c3c81-c16f-4673-84a9-82f8ac4034e0">

The same happens when tried to inject transient service into singleton as well.

**Comment by martincostello** [comment]
By _build_ it means when the service container is built when [`BuildServiceProvider()`](https://learn.microsoft.com/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectioncontainerbuilderextensions.buildserviceprovider) is called - it doesn't mean when you compile the application with `dotnet build`. It's behaving correctly by throwing when you run the application and the dependency graph is validated.
